src/exampleTypes/Date.js

```diff
@@ -3,22 +3,201 @@ import { observer } from 'mobx-react'
+    value: { from: 'now+1y/y', to: 'now+2y/y' },
+  },
+  {
+    type: 'future',
+    label: 'Next 30 Days',
+    value: { from: 'now/d', to: 'now/d+30d' },
+  },
+  {
+    type: 'future',
+    label: 'Next 12 Months',
+    value: { from: 'now/d', to: 'now/d+12M' },
+  },
+  {
+    type: 'future',
+    label: 'Next 24 Months',
+    value: { from: 'now/d', to: 'now/d+24M' },
+  },
+  {
+    type: 'future',
+    label: 'Next 36 Months',
+    value: { from: 'now/d', to: 'now/d+36M' },
+  },
+]
 
-let dateInput = x => <input type="date" {...x} />
 let DateComponent = injectTreeNode(
-  observer(({ tree, node, DateInput = dateInput }) => (
-    <Flex style={{ alignItems: 'center' }}>
-      <DateInput
-        value={node.from || ''}
-        onChange={e => tree.mutate(node.path, { from: e.target.value })}
-      />
-      <div>-</div>
-      <DateInput
-        value={node.to || ''}
-        onChange={e => tree.mutate(node.path, { to: e.target.value })}
-      />
-    </Flex>
-  )),
+  observer(
+    ({
+      tree,
+      node,
+      CustomPicker,
+      useJSPicker = true,
+      excludeRollingRanges = [],
+    }) => {
+      let Picker = CustomPicker || (useJSPicker ? DatePicker : html5DateInput)
+
+      let rollingOpts = _.reject(
+        opt => _.includes(opt.type, excludeRollingRanges),
+        allRollingOpts
+      )
+
+      let handleRollingSelection = idx => {
+        let range = rollingOpts[idx].value
+        tree.mutate(node.path, range)
+      }
+
+      return (
+        <div>
+          <RadioList
+            options={F.autoLabelOptions(['exact', 'rolling'])}
+            value={tree.getNode(node.path).useDateMath ? 'rolling' : 'exact'}
+            style={{ marginBottom: 10 }}
+            onChange={mode => {
+              tree.mutate(
+                node.path,
+                mode === 'rolling'
+                  ? {
+                      useDateMath: true,
+                      from: '',
+                      to: '',
+                    }
+                  : {
+                      useDateMath: false,
+                      from: null,
+                      to: null,
+                    }
+              )
+            }}
+          />
+          {!tree.getNode(node.path).useDateMath && (
+            <Flex
+              style={{ justifyContent: 'space-between', alignItems: 'center' }}
+            >
+              <Picker
+                value={node.from ? new Date(node.from) : null}
+                onChange={date => tree.mutate(node.path, { from: date })}
+                calendarType="US"
+              />
+              <div>-</div>
+              <Picker
+                value={node.to ? new Date(node.to) : null}
+                onChange={date => tree.mutate(node.path, { to: date })}
+                calendarType="US"
+              />
+            </Flex>
+          )}
+          {tree.getNode(node.path).useDateMath && (
+            <select onChange={e => handleRollingSelection(e.target.value)}>
```

Comment:
this should use the injected theme `Select` instead of assuming native

Reply:
we also need the value to be able to be dynamically determined by inspecting values on the node's from/to as we need to support loading from a pre-existing tree (e.g. a saved search)